### DISCUSSION TITLE
Skip tests on rez-build

### DISCUSSION THREAD

**Discussion by herronelou** [AcademySoftwareFoundation/rez#1334] [Q&A] [answered]
Hi, using rez-build with the python build system. My package has tests defined that run on build. In most cases, I want these tests to run, but in a few situations I want to be able to iterate quickly, and don't need all the tests to run. It doesn't seem like rez build has an argument to skip tests, is there a way to skip tests on build?

**Comment by ColinKennedy** [comment, marked as answer]
`rez-build` doesn't run any tests as a part of its work unless the tests are set to `"run_on": "pre_install"`, nor does Rez have a Python build system, so I'm not sure how you're these tests are initiating during your build. That said if you're using vanilla `rez-test`, you have some options

https://github.com/AcademySoftwareFoundation/rez/wiki/Package-Definition-Guide#tests

- explicit: Only run the test when asked (good for big test suites and / or auto-formatter code like `black` and `isort`)
- pre_release: Run only before release but not before regular `rez-build`. Though I think pre_release runs **before the build of the released package** (and not "after the build but before the release") so I think it won't do what you want it to.

If all else fails, just specifying the tests that you do want to run, explicitly, I think is the only other option.

**Comment by herronelou** [comment]
I do have the tests on pre-install. Maybe I should change them to
pre-release instead, and manually trigger the tests after install, but it's
fairly new that we have tests, so I wanted to make sure every other dev
runs the tests frequently. It wouldn't be so bad if it was a bit easier to
run tests (as #1335).

I'm also planning to run tests on merge requests via CI so maybe it's not
as critical as I first thought to have the tests on pre-install.
